### ISSUE TITLE
Crash on calling TakeDamageInfo with invalid damage info

### ISSUE THREAD

**Issue by mcNuggets1** [Facepunch/garrysmod-issues#3094]
### Details
Crashs when calling it.

### Steps to reproduce
Call Entity:TakeDamageInfo() with invalid damage info.

Resulting in crash.

**Comment by scnce** [comment]
I have experienced this same issue. I can provide some details.
The crash almost always occurs on Combine-type HL2 NPCs. Props and zombie-type HL2 NPCs sometimes don't cause a crash.
For example, entering:
`lua_run local info = DamageInfo() info:SetDamage(999) info:SetAttacker(Entity(1)) info:SetDamageType(DMG_GENERIC) Entity(1):GetEyeTrace().Entity:TakeDamageInfo(info)`
into the console while looking at a breakable crate or a zombie won't cause a crash. Entering it while looking at a combine soldier also won't usually crash the game. Entering this while looking at a strider, however, _will_ crash the game.
Entering:
`lua_run local info = DamageInfo() info:SetDamage(999) Entity(1):GetEyeTrace().Entity:TakeDamageInfo(info)`
while looking at a breakable crate or zombie will usually not crash the game, but entering it while looking at a combine soldier, hunter, strider, or any similar NPC **will always** instantly crash the game to desktop. I have tested this in both Windows and OS-X.
